[quant][fx] Add BatchNorm ops to backend_config_dict

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #75378
* #75377
* #75318
* #75317
* #75316
* #75315
* #75314
* __->__ #75260
* #75259
* #75258
* #75241

Summary:
This PR adds bn2d, bn3d and corresponding bn - relu to backend_config_dict

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35403587](https://our.internmc.facebook.com/intern/diff/D35403587)

jerryzh168 committed:
[wip][quant][fx] Add BatchNorm ops to backend_config_dict

Summary:
This PR adds bn2d, bn3d and corresponding bn - relu to backend_config_dict

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx] Add BatchNorm ops to backend_config_dict"


Summary:
This PR adds bn2d, bn3d and corresponding bn - relu to backend_config_dict

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][fx] Add BatchNorm ops to backend_config_dict"


Summary:
This PR adds bn2d, bn3d and corresponding bn - relu to backend_config_dict

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35403587).

andrewor14 commented on torch/ao/quantization/fx/backend_config/native.py:
```diff
@@ -351,6 +351,42 @@ def _get_binary_op_configs():
+        torch.nn.BatchNorm2d: nni.BNReLU2d,
+        torch.nn.BatchNorm3d: nni.BNReLU3d,
+    }
+    for bn in [torch.nn.BatchNorm2d, torch.nn.BatchNorm3d]:
```
Can you just iterate through `bn_to_fused_bn.keys()`?

jerryzh168 replied:
will fix in a future diff, I'm making some further changes down the road

andrewor14 commented on torch/ao/quantization/fx/backend_config/native.py:
```diff
@@ -351,6 +351,42 @@ def _get_binary_op_configs():
+            "dtype_configs": default_op_quint8_dtype_config,
+        })
+
+    for fused_bn in [nni.BNReLU2d, nni.BNReLU3d]:
```
Similarly, is this just `bn_to_fused_bn.values()`?

jerryzh168 committed:
Update on "[quant][fx] Add BatchNorm ops to backend_config_dict"


Summary:
This PR adds bn2d, bn3d and corresponding bn - relu to backend_config_dict

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35403587](https://our.internmc.facebook.com/intern/diff/D35403587)

[ghstack-poisoned]

jerryzh168 commented:
@jerryzh168 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D35403587).

jerryzh168 committed:
Update on "[quant][fx] Add BatchNorm ops to backend_config_dict"


Summary:
This PR adds bn2d, bn3d and corresponding bn - relu to backend_config_dict

Test Plan:
python test/test_quantization.py TestQuantizeFx
python test/test_quantization.py TestQuantizeFxOps

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D35403587](https://our.internmc.facebook.com/intern/diff/D35403587)

[ghstack-poisoned]
